docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,247 @@
+---
+layout: global
+title: TRANSFORM
+displayTitle: TRANSFORM
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `TRANSFORM` clause is used to specify a Hive-style transform (`SELECT TRANSFORM`/`MAP`/`REDUCE`)
+query specification to transform the input by running a specified script. Users can
+plug in their own custom mappers or reducers in the data stream by using features natively supported
+in the Spark SQL. In order to run a custom mapper script `map_script` or a custom
+reducer script `reduce_script` the user can issue the command which uses the `TRANSFORM`
+clause to embed the mapper or the reducer scripts.
+
+### Syntax
+
+```sql
+{ SELECT TRANSFORM ( named_expression [ , ... ] ) | { MAP | REDUCE } named_expression [ , ... ] }
```

Comment:
I'm still not sure if we should document MAP/REDUCE. It doesn't seem to match the Hive behavior and the naming is super weird if it's the same as TRANSFORM.

Reply:
> I'm still not sure if we should document MAP/REDUCE. It doesn't seem to match the Hive behavior and the naming is super weird if it's the same as TRANSFORM.

the different execution methods of Spark and hive make `MAP/REDUCE` same as `TRANSFORM`.

I think keep support about `MAP/REDUCE` is necessary. 
Since it's not totally match hive's behavior, we can remove this in document, tell user just use transform is ok.

Reply:
how about current?